### PR TITLE
basic draft

### LINKED ISSUE #3: Project setup
- [ ] multi env deploy
- [ ] BBF
- [ ] basic CORS
- [ ] Arch schema

### LINKED ISSUE #4: OAuth2 with Spotify

### LINKED ISSUE #7: Serve static files

### PR DESCRIPTION
**Pull request by Eomm**
closes #3 #7 #4

### CODE REVIEW

File: projects/frontend/src/features/Home.jsx [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
+import { useEffect, useState } from 'react'
+import { getHealth } from '../services/health'
+
+const API_URL = import.meta.env.VITE_API_URL
 
 export const Home = () => {
+  const [health, setHealth] = useState()
+
+  useEffect(() => {
+   getHealth().then((response) => setHealth(response))
```
Comment (Eomm) [inline projects/frontend/src/features/Home.jsx]: Should we add `setHealth` to the `useEffect` array?

File: projects/frontend/.env.sample [hunk 2]
```diff
@@ -0,0 +1 @@
+VITE_API_URL="http://localhost:3000"
```
Comment (Eomm) [inline projects/frontend/.env.sample:1]: Could you add this env var in the dockerfile/fly.toml to include it during the frontend build?

File: projects/backend/src/plugins/oauth.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+const oauthPlugin = require('@fastify/oauth2')
+
+module.exports = async function spotifyOauthPlugin (app, opts, next) {
```
Comment (Eomm) [inline projects/backend/src/plugins/oauth.js]: we choose one interface async or callback style

```suggestion
module.exports = async function spotifyOauthPlugin (app, opts) {
```

File: projects/backend/src/plugins/oauth.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+      auth: oauthPlugin.SPOTIFY_CONFIGURATION
+    },
+
+    startRedirectPath: '/login/spotify',
+    callbackUri: app.appConfig.OAUTH_REDIRECT_URI
+  })
+
+  app.register(require('@fastify/jwt'), {
+    secret: 'secret'
```
Comment (Eomm) [inline projects/backend/src/plugins/oauth.js]: the secret must be moved to an env var

File: projects/backend/src/plugins/oauth.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  })
+
+  app.decorate('authenticate', async function (request, reply) {
+    try {
+      await request.jwtVerify()
+    } catch (err) {
+      reply.send(err)
+    }
```
Comment (Eomm) [inline projects/backend/src/plugins/oauth.js]: when we use async, we should avoid `reply.send`

in this case, if the auth throw an error, the `errorHandler` will take care of it

```suggestion
    await request.jwtVerify()
```

File: projects/backend/src/plugins/oauth.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      rsh: encodeURIComponent(result.token.refresh_token),
+      name: data.display_name,
+      jti: data.email,
+      typ: 'sty',
+      exp: new Date(result.token.expires_at).getTime()
+    }
+
+    const token = app.jwt.sign({ payload })
+
+    console.log('JwtToken', token)
```
Comment (Eomm) [inline projects/backend/src/plugins/oauth.js]:
```suggestion
```

File: projects/frontend/src/features/Home.jsx (hunk 1 above)
Comment (AngyDev) [reply projects/frontend/src/features/Home.jsx]: We can do that but in general, the `setState` is stable, it never changes. This means that we can omit it from the `useEffect` or also `useCallback` dependency list.

**Review by AngyDev — APPROVED** [review]

**Pull request merged**